### PR TITLE
519/Edit Software of server

### PR DESCRIPTION
**Pull request by chrisma**
See: https://github.com/hpi-swt2/vm-portal/issues/519

### CODE REVIEW

**Comment by BjoernDaase** [comment]
I think we should also give the possibility to either remove software when editing or at least delete the empty fields. Otherwise, they will grow over time without having content.

**Comment by chrisma** [comment]
> I think we should also give the possibility to either remove software when editing or at least delete the empty fields. Otherwise, they will grow over time without having content.

Software can be removed by visiting the `Software#edit` page and removing the rendered string from the input. You are correct, empty form fields are currently not removed in the controller. I didn' feel like investing any more work into a "feature" that should have been a single text field.

**Comment by chrisma** [comment]
I fixed it anyways

File: app/controllers/servers_controller.rb [hunk 1]
```diff
@@ -23,114 +22,44 @@ def new
...
   # Use callbacks to share common setup or constraints between actions.
   def set_server
     @server = Server.find(params[:id])
   end
 
   # Never trust parameters from the scary internet, only allow the white list through.
   def server_params
-    params.fetch(:server, {})
+    # Remove empty software form fields
+    params[:server][:installed_software].reject!(&:blank?) if params[:server][:installed_software]
```
Comment (MrModder) [inline app/controllers/servers_controller.rb]: Won't this crash if params[:server] is nil?
We should to check for this in the condition

Comment (chrisma) [reply app/controllers/servers_controller.rb]: > Won't this crash if `params[:server]` is nil?

Yes. If no `server` param is given and this line wasn't there, there would be an error one line later in the `require` call: https://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-require

Comment (chrisma) [reply app/controllers/servers_controller.rb]: But good point, the original exception is neater. I've dug around in `ActionController::Parameters` and found an acceptable way of changing params after permit.

**Comment by BjoernDaase** [comment]
@chrisma @MrModder I resolved the conflicts so this should be ready to merge now, right?

**Pull request merged**